beacon-chain/state/stategen/cold.go

```diff
@@ -37,6 +37,66 @@ func (s *State) saveColdState(ctx context.Context, blockRoot [32]byte, state *st
 	return nil
 }
 
+// This loads the cold state by block root, it decides whether to load from archived point (faster) or
+// somewhere between archived points (slower) because it requires replaying blocks.
+// This method is more efficient than load cold state by slot.
+func (s *State) loadColdStateByRoot(ctx context.Context, blockRoot [32]byte) (*state.BeaconState, error) {
```

Comment:
Thoughts on metrics for this function? Seems like an interesting candidate down the line (doesn't have to be in this PR)

Reply:
yea will add that tomorrow, thanks!